RapidReact/src/main/java/frc/robot/subsystem/ClimberSubsystem.java

```diff
@@ -1,112 +1,311 @@
+
+    climber1.set(ControlMode.PercentOutput, -climbOutput.currentValue());
+    climber2.set(ControlMode.PercentOutput, -climbOutput.currentValue());
+    climberMotorPosition.log(climber1.getSelectedSensorPosition());
 
     climbState.log(LogLevel.GENERAL, "elevatorRetract");
   }
 
   public void elevatorStop() {
-    climber.set(0);
+    climber1.set(0);
+    climber2.set(0);
     climbState.log(LogLevel.GENERAL, "climbStopped");
   }
 
-  public void climbAuto() { //uses TPAD button
+  // autoClimb() both enables autoclimb and sets the flag to true
+  // autoClimbReleased() just sets the flag to false
+  public void autoClimb() { //uses TPAD button
     autoClimb = true;
-    climbState.log(LogLevel.GENERAL, "climb2auto");
+    autoClimbPressed = true;
+
+    climbState.log(LogLevel.GENERAL, "automatically climbing");
+  }
+  public void autoClimbReleased()
+  {
+    autoClimbPressed = false;
   }
 
-  public void elevatorToggle() {
-    if (config.enablePneumatics) {
-      elevatorToggle = !elevatorToggle;
-      if (!elevatorToggle) {
-        elevatorSolenoid.set(Value.kForward);
-        elevatorToggleState.log(LogLevel.GENERAL, "ClimbTime");
-      } else {
-        elevatorSolenoid.set(Value.kReverse);
-        elevatorToggleState.log(LogLevel.GENERAL, "NotClimbTime");
-      }
+  private void autoExtendPartial()
+  {
+    climber1.set(TalonSRXControlMode.MotionMagic, partialExtendPosition);
+    climber2.set(TalonSRXControlMode.MotionMagic, partialExtendPosition);
+
+    climberMotorPosition.log(climber1.getSelectedSensorPosition());
+    climbState.log(LogLevel.GENERAL, "elevatorExtendPartial");
+  }
+
+  private void autoExtendFull()
+  {
+    climber1.set(TalonSRXControlMode.MotionMagic, fullExtendPosition);
+    climber2.set(TalonSRXControlMode.MotionMagic, fullExtendPosition);
+
+    climberMotorPosition.log(climber1.getSelectedSensorPosition());
+    climbState.log(LogLevel.GENERAL, "elevatorExtendFull");
+  }
+
+  private void autoRetractFull()
+  {
+    climber1.set(TalonSRXControlMode.MotionMagic, fullRetractPosition);
+    climber2.set(TalonSRXControlMode.MotionMagic, fullRetractPosition);
+
+    climberMotorPosition.log(climber1.getSelectedSensorPosition());
+    climbState.log(LogLevel.GENERAL, "elevatorRetract");
+  }
+
+  // TODO: add a button for this
+  private void stopAutoClimb() {
+    autoClimb = false;
+    currentClimbState = ClimbState.Idle;
+  }
+
+  public void setElevatorTilted(boolean tilted)
+  {
+    if (!config.enablePneumatics)
+      return;
+    
+    climberTilted = tilted;
+    
+    if (tilted)
+    {
+      elevatorSolenoid.set(Value.kReverse);
+      elevatorTiltedState.log(LogLevel.GENERAL, "Tilted");
+    }
+    else
+    {
+      elevatorSolenoid.set(Value.kForward);
+      elevatorTiltedState.log(LogLevel.GENERAL, "Not Tilted");
     }
   }
 
+  public void elevatorToggle()
```

Comment:
toggle functions bad! use set functions only!

Reply:
yeah this is just for backwards compatibility with the control scheme until we get everyone to agree that we want different buttons to toggle it back and forth